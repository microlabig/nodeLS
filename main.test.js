/* eslint-disable */
const runCommand = require("./utils");
const dirTree = require("directory-tree");

const mock = require("mock-fs");

const APP_PATH = require.resolve("./main");

describe("app: file system", () => {
    beforeEach(() => {
        mock({
            './src': mock.directory({
                items: {
                    from: {
                        folder1: {
                            'file1.txt': "text content 1",
                            'file2.txt': "text content 2"
                        },
                        folder2: {
                            inner_folder: {
                                'file3.txt': "text content 3"
                            }
                        },
                        'file0.txt': "text content 0"
                    }
                }

            })
        })
    });
    afterEach(mock.restore);

    it("create snapshot", async () => {
        const stdout = await runCommand(`${APP_PATH} from to --d`);

        mock.restore();
        expect(stdout).toMatchSnapshot();
    });

    it("creates sorted files", async () => {
        await runCommand(`${APP_PATH} from to --d`);
        const tree = dirTree("./src");

        mock.restore();
        console.log(tree);

        expect(tree).toMatchSnapshot();
    });
});
