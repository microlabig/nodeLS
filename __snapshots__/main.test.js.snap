// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app: file system create snapshot 1`] = `""`;

exports[`app: file system creates sorted files 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "extension": ".txt",
          "name": "file0.txt",
          "path": "src\\\\from\\\\file0.txt",
          "size": 14,
          "type": "file",
        },
        Object {
          "children": Array [
            Object {
              "extension": ".txt",
              "name": "file1.txt",
              "path": "src\\\\from\\\\folder1\\\\file1.txt",
              "size": 14,
              "type": "file",
            },
            Object {
              "extension": ".txt",
              "name": "file2.txt",
              "path": "src\\\\from\\\\folder1\\\\file2.txt",
              "size": 14,
              "type": "file",
            },
          ],
          "name": "folder1",
          "path": "src\\\\from\\\\folder1",
          "size": 28,
          "type": "directory",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "extension": ".txt",
                  "name": "file3.txt",
                  "path": "src\\\\from\\\\folder2\\\\inner_folder\\\\file3.txt",
                  "size": 14,
                  "type": "file",
                },
              ],
              "name": "inner_folder",
              "path": "src\\\\from\\\\folder2\\\\inner_folder",
              "size": 14,
              "type": "directory",
            },
          ],
          "name": "folder2",
          "path": "src\\\\from\\\\folder2",
          "size": 14,
          "type": "directory",
        },
      ],
      "name": "from",
      "path": "src\\\\from",
      "size": 56,
      "type": "directory",
    },
  ],
  "name": "src",
  "path": "./src",
  "size": 56,
  "type": "directory",
}
`;
